[ci] MockTcpTransportTests.testNotifyOnShutdown

Doesn't reproduce locally on master

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=oraclelinux/2617/console

[build-2617.txt](https://github.com/elastic/elasticsearch/files/2288774/build-2617.txt)

```
ERROR   0.10s J0 | MockTcpTransportTests.testNotifyOnShutdown <<< FAILURES!
   > Throwable #1: com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=432, name=elasticsearch[org.elasticsearch.transport.MockTcpTransportTests][generic][T#1], state=RUNNABLE, group=TGRP-MockTcpTransportTests]
   >    at __randomizedtesting.SeedInfo.seed([CBC399AED8F96471:26E336404194B6E]:0)
   > Caused by: java.lang.AssertionError: Can't move to stopped state when closed
   >    at __randomizedtesting.SeedInfo.seed([CBC399AED8F96471]:0)
   >    at org.elasticsearch.transport.AbstractSimpleTransportTestCase.lambda$testNotifyOnShutdown$10(AbstractSimpleTransportTestCase.java:775)
   >    at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
   >    at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1497)
   >    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:723)
   >    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   >    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   >    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   >    at java.lang.Thread.run(Thread.java:748)
```

```
./gradlew :test:framework:test -Dtests.seed=CBC399AED8F96471 -Dtests.class=org.elasticsearch.transport.MockTcpTransportTests -Dtests.method="testNotifyOnShutdown" -Dtests.security.manager=true -Dtests.locale=el-CY -Dtests.timezone=IST -Dcompiler.java=10 -Druntime.java=8
```

--- a/test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java
@@ -768,6 +768,7 @@ public abstract class AbstractSimpleTransportTestCase extends ESTestCase {
 
     public void testNotifyOnShutdown() throws Exception {
         final CountDownLatch latch2 = new CountDownLatch(1);
+        final CountDownLatch latch3 = new CountDownLatch(1);
         try {
             serviceA.registerRequestHandler("internal:foobar", StringMessageRequest::new, ThreadPool.Names.GENERIC,
                 (request, channel, task) -> {
@@ -777,6 +778,8 @@ public abstract class AbstractSimpleTransportTestCase extends ESTestCase {
                         serviceB.stop();
                     } catch (Exception e) {
                         fail(e.getMessage());
+                    } finally {
+                        latch3.countDown();
                     }
                 });
             TransportFuture<TransportResponse.Empty> foobar = serviceB.submitRequest(nodeA, "internal:foobar",
@@ -788,6 +791,7 @@ public abstract class AbstractSimpleTransportTestCase extends ESTestCase {
             } catch (TransportException ex) {
 
             }
+            latch3.await();
         } finally {
             serviceB.close(); // make sure we are fully closed here otherwise we might run into assertions down the road
             serviceA.disconnectFromNode(nodeB);

